### PR TITLE
Fix formatting in all files

### PR DESCRIPTION
**Pull request by nohwnd**
Fixed formatting in all files and fixed tests that rely on formatting

### CODE REVIEW

File: src/Pester.psd1 [hunk 1]
```diff
@@ -95,7 +95,7 @@
         # It's for rebuilding PowerShellGet (and PoshCode) NuGet-style packages
         # We had to do this because it's the only place we're allowed to extend the manifest
         # https://connect.microsoft.com/PowerShell/feedback/details/421837
-        PSData = @{
+        PSData                  = @{
```
Comment (nohwnd) [inline src/Pester.psd1:98]: the PSD1 formatting of hashtables is weird, but I am okay with that.

File: PSScriptAnalyzerSettings.psd1 [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
-    Severity = @('Error','Warning')
+    Severity            = @('Error', 'Warning')
     IncludeDefaultRules = $true
-    CustomRulePath = './Pester.BuildAnalyzerRules'
-    ExcludeRules=@(
+    CustomRulePath      = './Pester.BuildAnalyzerRules'
+    ExcludeRules        = @(
         'PSUseShouldProcessForStateChangingFunctions'
         'PSUseApprovedVerbs'
         'Measure-SafeCommands'
```
Comment (nohwnd) [inline PSScriptAnalyzerSettings.psd1:8]: Same here. PSD1 formatting sucks.

**Comment by fflaten** [comment]
Since we're enabling format on save here, should we standardize on UTF8-bom encoding for ps1/psm1 files at the same time? Related #1747

With BOM due to Windows PowerShell versions.

**Comment by nohwnd** [comment]
Sure.  Moving discussion to the issue.

**Pull request merged**